specs/core/0_beacon-chain.md

```diff
@@ -491,6 +510,15 @@ def get_block_hash(state: BeaconState,
 
 `get_block_hash(_, _, s)` should always return the block hash in the beacon chain at slot `s`, and `get_shards_and_committees_for_slot(_, s)` should not change unless the validator set changes.
 
+The following is a function that determines the proposer of a beacon block:
+
+```python
+def get_beacon_proposer(state, slot):
+    first_committee = get_shards_and_committees_for_slot(state, slot)[0]
```

Comment:
Note that in Python implement, we'll handle the case when `get_shards_and_committees_for_slot(state, slot)` is an empty list.

Reply:
can't ever be empty unless the chain stops! but then there will be no proposers :)